loadtest/loadtest.go

```diff
@@ -47,7 +59,20 @@ func (lt *LoadTester) handleStatus() {
 
 // AddUser increments by one the number of concurrently active users.
 func (lt *LoadTester) AddUser() error {
-	if !lt.started {
+	lt.mut.Lock()
+	defer lt.mut.Unlock()
+
+	return lt.addUser(false)
+}
+
+// addUser is an internal API called from Run and AddUser both.
+// DO NOT call this by itself, because this method is not protected by a mutex.
+// isStarting is true when called from Start, and false when called from
+// AddUser.
+func (lt *LoadTester) addUser(isStarting bool) error {
```

Comment:
Why do we need to pass this `bool` value? Can't we just check the state?

Reply:
Yes, but this is more precise. Just checking for state would broaden the allowed states to 2. But with this, it can only be in one state.